sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs

```diff
@@ -318,6 +318,110 @@ public virtual async Task<Response<LogsBatchQueryResultCollection>> QueryBatchAs
+        /// {
+        ///     Console.WriteLine(&quot;Name: &quot; + columns.Name + &quot; Type: &quot; + columns.Type);
+        /// }
+        /// </code>
+        /// </example>
+        /// </summary>
+        /// <param name="resourceId"> The resourceId where the query should be executed. </param>
+        /// <param name="query"> The Kusto query to fetch the logs. </param>
+        /// <param name="timeRange"> The time period for which the logs should be looked up. </param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>The logs matching the query.</returns>
+        public virtual Response<LogsQueryResult> QueryResource(ResourceIdentifier resourceId, string query, QueryTimeRange timeRange, CancellationToken cancellationToken = default)
+        {
+            string resource = resourceId.ToString();
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(LogsQueryClient)}.{nameof(QueryResource)}");
+            scope.Start();
+            try
+            {
+                while (resource.StartsWith("/"))
+                {
+                    resource = resource.Substring(1);
+                }
+                return _queryClient.ResourceGet(resource, query, timeRange.Duration, cancellationToken);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Returns all the Azure Monitor logs matching the given query for an Azure resource.
+        /// <example snippet="Snippet:QueryResource">
+        /// <code language="csharp">
+        /// var client = new LogsQueryClient(new DefaultAzureCredential());
+        ///
+        /// var results = await client.QueryResourceAsync(TestEnvironment.StorageAccountId,
+        ///     &quot;search *&quot;,
+        ///     new QueryTimeRange(TimeSpan.FromDays(5)));
+        ///
+        /// var resultTable = results.Value.Table;
+        ///
+        /// foreach (LogsTableRow rows in resultTable.Rows)
+        /// {
+        ///     foreach (var row in rows)
+        ///     {
+        ///         Console.WriteLine(row);
+        ///     }
+        /// }
+        ///
+        /// foreach (LogsTableColumn columns in resultTable.Columns)
+        /// {
+        ///     Console.WriteLine(&quot;Name: &quot; + columns.Name + &quot; Type: &quot; + columns.Type);
+        /// }
+        /// </code>
+        /// </example>
+        /// </summary>
+        /// <param name="resourceId"></param>
+        /// <param name="query"></param>
+        /// <param name="timeRange"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public virtual async Task<Response<LogsQueryResult>> QueryResourceAsync(ResourceIdentifier resourceId, string query, QueryTimeRange timeRange, CancellationToken cancellationToken = default)
```

Comment:
We should add another overload `QueryResourceAsync<T>` that returns `Azure.Response<System.Collections.Generic.IReadOnlyList<T>>`